packages/sdk/src/wallet/DynamicWallet.ts

```diff
@@ -0,0 +1,104 @@
+import { isEthereumWallet } from '@dynamic-labs/ethereum'
+import type { DynamicWaasEVMConnector } from '@dynamic-labs/waas-evm'
+import {
+  type Address,
+  createWalletClient,
+  fallback,
+  http,
+  type LocalAccount,
+  type WalletClient,
+} from 'viem'
+import { toAccount } from 'viem/accounts'
+
+import type { SupportedChainId } from '@/constants/supportedChains.js'
+import type { ChainManager } from '@/services/ChainManager.js'
+import type { DynamicHostedWalletToVerbsWalletOptions } from '@/types/wallet.js'
+import { Wallet } from '@/wallet/base/Wallet.js'
+
+/**
+ * Dynamic wallet implementation
+ * @description Wallet implementation using the Dynamic Labs wallet SDK.
+ */
+export class DynamicWallet extends Wallet {
+  public signer!: LocalAccount
+  public address!: Address
+  private readonly dynamicWallet: DynamicHostedWalletToVerbsWalletOptions['wallet']
+
+  /**
+   * Create a new Dynamic wallet
+   * @param chainManager - Chain manager for RPC, chain info, and transports
+   * @param dynamicWallet - Dynamic Labs wallet instance (EVM)
+   */
+  private constructor(
+    chainManager: ChainManager,
+    dynamicWallet: DynamicHostedWalletToVerbsWalletOptions['wallet'],
+  ) {
+    super(chainManager)
+    this.dynamicWallet = dynamicWallet
```

Comment:
🤔 If we renamed this something general like `this.hostedWallet`, can it be abstracted into the parent? If so, would it reduce the amount of boilerplate for this and and PrivyWallet?